modules/navigation/3d/nav_map_builder_3d.cpp

```diff
@@ -0,0 +1,398 @@
+				continue;
+			}
+
+			// Check if the two edges are close to each other enough and compute a pathway between the two regions.
+			Vector3 self1 = edge_vector * CLAMP(projected_p1_ratio, 0.0, 1.0) + edge_p1;
+			Vector3 other1;
+			if (projected_p1_ratio >= 0.0 && projected_p1_ratio <= 1.0) {
+				other1 = other_edge_p1;
+			} else {
+				other1 = other_edge_p1.lerp(other_edge_p2, (1.0 - projected_p1_ratio) / (projected_p2_ratio - projected_p1_ratio));
+			}
+			if (other1.distance_squared_to(self1) > edge_connection_margin_squared) {
+				continue;
+			}
+
+			Vector3 self2 = edge_vector * CLAMP(projected_p2_ratio, 0.0, 1.0) + edge_p1;
+			Vector3 other2;
+			if (projected_p2_ratio >= 0.0 && projected_p2_ratio <= 1.0) {
+				other2 = other_edge_p2;
+			} else {
+				other2 = other_edge_p1.lerp(other_edge_p2, (0.0 - projected_p1_ratio) / (projected_p2_ratio - projected_p1_ratio));
+			}
+			if (other2.distance_squared_to(self2) > edge_connection_margin_squared) {
+				continue;
+			}
+
+			// The edges can now be connected.
+			gd::Edge::Connection new_connection = other_edge;
+			new_connection.pathway_start = (self1 + other1) / 2.0;
+			new_connection.pathway_end = (self2 + other2) / 2.0;
+			free_edge.polygon->edges[free_edge.edge].connections.push_back(new_connection);
+
+			// Add the connection to the region_connection map.
+			map_iteration->external_region_connections[(uint32_t)free_edge.polygon->owner->id].push_back(new_connection);
+			performance_data.pm_edge_connection_count += 1;
+		}
+	}
+}
+
+void NavMapBuilder3D::_build_step_navlink_connections(NavMapIterationBuild &r_build) {
+	NavMapIteration *map_iteration = r_build.map_iteration;
+	const Vector3 &merge_rasterizer_cell_size = r_build.merge_rasterizer_cell_size;
+	real_t link_connection_radius = r_build.link_connection_radius;
+	real_t link_connection_radius_sqr = link_connection_radius * link_connection_radius;
+	int polygon_count = r_build.polygon_count;
+	int link_polygons_count = r_build.link_polygons_count;
+
+	// Search for polygons within range of a nav link.
+	for (NavLinkIteration &link : map_iteration->link_iterations) {
+		if (!link.enabled) {
+			continue;
+		}
+		const Vector3 link_start_pos = link.start_position;
+		const Vector3 link_end_pos = link.end_position;
+
+		gd::Polygon *closest_start_polygon = nullptr;
+		real_t closest_start_sqr_dist = link_connection_radius_sqr;
+		Vector3 closest_start_point;
+
+		gd::Polygon *closest_end_polygon = nullptr;
+		real_t closest_end_sqr_dist = link_connection_radius_sqr;
+		Vector3 closest_end_point;
+
+		for (NavRegionIteration &region : map_iteration->region_iterations) {
+			AABB region_bounds = region.bounds.grow(link_connection_radius);
+			if (!region_bounds.has_point(link_start_pos) && !region_bounds.has_point(link_end_pos)) {
+				continue;
+			}
+			for (gd::Polygon &polyon : region.navmesh_polygons) {
```

Comment:
No because we also need the 2 closest polygon pointers. The polygons_get_closest_point_info () function gives only the owner RID which does not help in this case.